pytest_html/plugin.py

```diff
@@ -105,12 +105,51 @@ def __init__(self, outcome, report, logfile, config):
             self.logfile = logfile
             self.config = config
             self.row_table = self.row_extra = None
+            self.images = []
 
             test_index = hasattr(report, 'rerun') and report.rerun + 1 or 0
 
             for extra_index, extra in enumerate(getattr(report, 'extra', [])):
                 self.append_extra_html(extra, extra_index, test_index)
 
+            num_of_images = len(self.images)
+            if num_of_images == 1:  # Only one image, no need for carousel
+                src, name = self.images.pop()
+                self.additional_html.append(
+                    html.div(html.a(html.img(src=src, alt=name),
+                                    href=src, target='_blank'),
+                             class_='image'))
+
+            elif num_of_images > 1:  # More than one image, build carousel
+                image_html = []
+                for image in range(num_of_images):
```

Comment:
```
for i in range(iterable):
    iterable[i]
```
Doesn't look very pythonic
Have a look here: https://stackoverflow.com/questions/1011938/python-previous-and-next-values-inside-a-loop